lib/evaluateFunctions.js

```diff
@@ -96,4 +99,389 @@ function absoluteValue(node) {
+  if (status.hasChanged()) {
+    substeps.push(status);
+    newNode = NodeStatus.resetChangeGroups(status.newNode);
+
+    status = combineRoots(newNode);
+    if (status.hasChanged()) {
+      substeps.push(status);
+      newNode = NodeStatus.resetChangeGroups(status.newNode);
+    }
+
+    return NodeStatus.nodeChanged(
+      MathChangeTypes.NTH_ROOT_VALUE, node, newNode, true, substeps);
+  }
+
+  return NodeStatus.noChange(node);
+}
+
+// Given an nthRoot node with a constant positive root, will do the step of
+// factoring all the arguments under the radicand
+function factorChildren(node) {
+  let newNode = clone(node, false);
+  const radicandNode = node.args[0];
+
+  let children = [];
+  for (let i = 0; i < radicandNode.args.length; i++) {
+    const child = radicandNode.args[i];
+
+    if (NodeType.isConstant(child) && !Negative.isNegative(child)) {
+      const radicandValue = parseFloat(child.value);
+      const factors = Factor.getPrimeFactors(radicandValue);
+      const factorNodes = factors.map(NodeCreator.constant);
+      children = children.concat(factorNodes);
+    }
+    else {
+      children.push(child);
+    }
+  }
+
+  if (children.length !== radicandNode.args.length) {
+    newNode.args[0] = NodeCreator.operator('*', children);
+    return NodeStatus.nodeChanged(
+      MathChangeTypes.FACTOR_INTO_PRIMES, node, newNode);
+  }
+
+  return NodeStatus.noChange(node);
+}
+
+// Given an nthRoot node with a constant positive root, will group the arguments
+// into groups of the root as a step
+// e.g. nthRoot(2 * 2 * 2, 2) -> nthRoot((2 * 2) * 2, 2)
+function groupTermsByRoot(node) {
+  let newNode = clone(node, false);
+  const radicandNode = node.args[0];
+  const rootNode = node.args.length === 2 ? node.args[1] : NodeCreator.constant(2);
+  const rootValue = parseFloat(rootNode.value);
+
+  radicandNode.args.sort((a, b) => {
+    if (NodeType.isConstant(a) && NodeType.isConstant(b)) {
+      return parseFloat(a.value) - parseFloat(b.value);
+    }
+    else if (NodeType.isConstant(a)) {
+      return -1;
+    }
+    else if (NodeType.isConstant(b)) {
+      return 1;
+    }
+    return 0;
+  });
+
+  let children = [], hasGroups = false;
+  for (let i = 0; i < radicandNode.args.length;) {
+    let j = i;
+    for (; j < radicandNode.args.length; j++) {
+      if (!radicandNode.args[i].equals(radicandNode.args[j]) || (j-i) == rootValue) {
+        break;
+      }
+    }
+    if (j - i === rootValue) {
+      hasGroups = true;
+      const groupedNode = NodeCreator.parenthesis(
+        NodeCreator.operator('*', radicandNode.args.slice(i, j)));
+      children.push(groupedNode);
+    }
+    else {
+      children = children.concat(radicandNode.args.slice(i, j));
+    }
+    i = j;
+  }
+
+  // if we don't group any factors, then we can't simplify it any more
+  if (hasGroups) {
+    newNode.args[0] = children.length === 1 ?
+        children[0] : NodeCreator.operator('*', children);
+    return NodeStatus.nodeChanged(
```

Comment:
did you go through all your changes and see if they need change groups? or are you leaving that for your other task